lib/query/lib/prepareForProcess.js

```diff
@@ -44,12 +44,40 @@ function applyPagination(body, _params) {
     }
 }
 
-export default (_body, _params = {}) => {
+function handleTextFilters(data, isClient) {
+    _.each(data, field => {
+        if (field.$text) {
+            if (isClient) {
+                let $or = _.map(field.$text.$fields, (val, key) => {
+                    return {
+                        [key]: {
+                            $regex:field.$text.$search,
```

Comment:
The search string should have some regex added before and after it that matches beginning-of-line, end-of-line, space and any other character mongo counts as a word delimiter, so that it more closely matches the behavior of a mongo text search.